src/toolchain/toolchain.ts

```diff
@@ -178,6 +187,47 @@ export class SwiftToolchain {
         return xcodes.trimEnd().split("\n");
     }
 
+    /**
+     * Get a list of new project templates from swift package manager
+     * @returns a {@link SwiftProjectTemplate} for each discovered project type
+     */
+    public async getProjectTemplates(): Promise<SwiftProjectTemplate[]> {
+        const { stdout } = await execSwift(["package", "init", "--help"], "default");
+        const lines = stdout.split(/\r?\n/g);
```

Comment:
This is some funky way to get project templates, but yeah can't see any other way of doing it.

I almost wonder if it is worthwhile rolling our own outside of SwiftPM. Would love to give users the ability to create their own templates. Maybe using Mustache templating to allow users to add in customisation. But that'd be for another task I think.

Reply:
Yeah I agree. I'd honestly like an option in Swift Package Manager to get these as JSON rather than doing the crazy command line parsing. However, we still need to support Swift versions <6.